cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/NodeDAO.java

```diff
@@ -584,65 +584,46 @@ public void getChild(ContainerNode parent, String name, boolean resolveMetadata)
                 catch(Throwable oops) { log.error("failed to rollback dirtyRead transaction in finally", oops); }
         }
     }
-
-    /**
-     * Load all the child nodes of the specified container.
-     *
-     * @param parent
-     */
-    public void getChildren(ContainerNode parent) throws TransientException
-    {
-        log.debug("getChildren: " + parent.getUri().getPath() + ", " + parent.getClass().getSimpleName());
-        getChildren(parent, null, null, true);
+    
+    public void getChildren(ContainerNode parent) throws TransientException {
+        this.getChildren(parent, null, null, null, true, true);
     }
-
-    /**
-     * Load all the children of a container based on whether the client 
-     * wants the metadata to be resolved.
-     *
-     * @param parent
-     * @param resolveMetadata
-     */
-    public void getChildren(ContainerNode parent, boolean resolveMetadata) throws TransientException
-    {
-        log.debug("getChildren: " + parent.getUri().getPath() + ", " + parent.getClass().getSimpleName());
-        getChildren(parent, null, null, resolveMetadata);
+    
+    public void getChildren(ContainerNode parent,  VOSURI start, Integer limit) throws TransientException {
+        this.getChildren(parent, start, limit, null, true, true);
     }
 
     /**
-     * Loads some of the child nodes of the specified container.
-     * @param parent
-     * @param start
-     * @param limit
-     */
-    public void getChildren(ContainerNode parent, VOSURI start, Integer limit) throws TransientException
-    {
-        log.debug("getChildren: " + parent.getUri().getPath() + ", " + parent.getClass().getSimpleName());
-        getChildren(parent, start, limit, true);
-    }
-    
-    /**
-     * Loads some of the child nodes of the specified container based on 
-     * whether the client wants the metadata to be resolved.
+     * 
      * @param parent
      * @param start
      * @param limit
+     * @param sortProperty
+     * @param sortAsc
      * @param resolveMetadata
+     * @throws TransientException
      */
-    public void getChildren(ContainerNode parent, VOSURI start, Integer limit, boolean resolveMetadata) throws TransientException
+    public void getChildren(ContainerNode parent,  VOSURI start, Integer limit, URI sortProperty, Boolean sortAsc, boolean resolveMetadata) throws TransientException
     {
         log.debug("getChildren: " + parent.getUri().getPath() + ", " + parent.getClass().getSimpleName());
         expectPersistentNode(parent);
 
         Object[] args = null;
-        if (start != null)
+        if (start != null) {
             args = new Object[] { start.getName() };
-        else
+            if (sortProperty != null) {
+                // get the persistent node
+                Node startNode = this.getPath(start.getPath(), true, false);
+                args = new Object[] { startNode.getPropertyValue(sortProperty.toString()) };
```

Comment:
The startNode here can be null if the start node was deleted (by client A) while iterating (by client B). This didn't use to matter when iterating by name but now it does. The question is: what to do when you cannot get the sort property anymore? I think the request has to fail... TBD.